### ISSUE TITLE
ValueError During Training Process

### ISSUE THREAD

**Issue by manhph2211** [Wadaboa/titanet#1]
Dear sir, 

First, I have to say that I really appreciate that you publish the code in this repo. What an amazing implementation! 
However, I've just tried to train the given dataset followed by your tutorial and it seemed to work well until I saw the error:
![image](https://user-images.githubusercontent.com/61444616/140864070-a77dfc9a-a720-4b01-89c1-468bdffa75ff.png)

I realized that final batch is really important for tasks related to the used custom dataset ! I think the error comes from the process of final batch. 

Please give me feedback on this, and hopefully you will update the code 😄

**Comment by Wadaboa** [comment]
Hi there,

Thank you very much for noticing the issue.

As you pointed out, the error is raised by the batch normalization layer when the input batch only has 1 example. To fix the issue, you can try to either change the batch size (in the `parameters.yml` file) so that `len(dataset) % batch_size != 1` (where `%` means modulo) or you can add `drop_last=True` in the Dataloader creation phase.

Would you mind trying these solutions out and letting me know if they actually fix the issue? If they do, you can also file a PR or I can commit the changes, as you wish.

Thanks,
Alessio.

**Comment by manhph2211** [comment]
Hi Wadaboa, 

I'm sorry for the late response. Your solutions should fix the issue well 😄  Just update changes ❤️ 

Hope you have a great day!

Max

**Comment by Wadaboa** [comment]
Fixed in b07e3074e79ea8c1129fb0adb8315e06bb4943ea, closing the issue.
